### PR TITLE
added

### PR DESCRIPTION
**Pull request by Truput**

### CODE REVIEW

File: .idea/.gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Default ignored files
+/shelf/
```
Comment (voljabes) [inline .idea/.gitignore:2]: Consider to change it according to this one https://www.toptal.com/developers/gitignore/api/jetbrains

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,2 @@
+# Project exclude paths
```
Comment (voljabes) [inline .gitignore:1]: Please  consider to change it according with this one https://www.toptal.com/developers/gitignore/api/java

Comment (voljabes) [inline .gitignore:1]: Consider to remove unneeded files from this branch, e.g. log ones, etc.

File: src/test/java/tests/Hardcore.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.testng.Assert;
+import org.testng.annotations.AfterSuite;
+import org.testng.annotations.BeforeSuite;
+import org.testng.annotations.Test;
+import pages.CloudGoogleHome;
+import pages.EstimateCompletePage;
+import pages.TemporaryEmailPage;
+
+public class Hardcore {
```
Comment (voljabes) [inline src/test/java/tests/Hardcore.java:14]: I would like to suggest to rename it to HardwareTests or VirtualMashineTest or CostEstimationTests

File: src/test/java/tests/Hardcore.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+    @Test
+    public void estimatedCostInEmailMatchWithCalculator() {
+        double estimatedCostInCalculator = estimateCompletePage.getTotalEstimatedCost();
+        estimateCompletePage.clickEmailEstimateButton();
+        TemporaryEmailPage temporaryEmailPage = estimateCompletePage.createNewTabAndSwitch();
+        temporaryEmailPage.openPage().copyGeneratedEmail().switchToTab();
+        estimateCompletePage.inputEmail().sendEmail().switchToTab();
+        temporaryEmailPage.openEmail();
+        double estimatedCostInEmail = temporaryEmailPage.estimatedCost();
```
Comment (voljabes) [inline src/test/java/tests/Hardcore.java:45]: Consider to rename method to getEstimatedCost

File: src/test/java/tests/Hardcore.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        driver = null;
+    }
+
+
+    private void estimateGoogleCloud() {
+        estimateCompletePage = new CloudGoogleHome(driver)
+                .openPage()
+                .searchByTerm(SEARCH_TERM)
+                .openResultLink()
+                .enterTermToEngineField(ENGIN_FIELD_SEARCH_TERM)
```
Comment (voljabes) [inline src/test/java/tests/Hardcore.java:62]: Engine

File: src/test/java/tests/HurtMePlenty.java [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+    @BeforeSuite(alwaysRun = true)
+    public void browserSetup() {
+        driver = new ChromeDriver();
+        driver.manage().window().maximize();
+        estimateGoogleCloud();
+    }
+
+    @Test
+    public void vmClassMathes() {
+        String vmClass = estimateCompletePage.getVmClass();
```
Comment (voljabes) [inline src/test/java/tests/HurtMePlenty.java:39]: Consider to rename it to expectedVMClass

File: src/test/java/tests/HurtMePlenty.java [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+
+    @BeforeSuite(alwaysRun = true)
+    public void browserSetup() {
+        driver = new ChromeDriver();
+        driver.manage().window().maximize();
+        estimateGoogleCloud();
+    }
+
+    @Test
+    public void vmClassMathes() {
```
Comment (voljabes) [inline src/test/java/tests/HurtMePlenty.java:38]: I wonder if method might be renamed into some other name that not uses 'matches''. if one uses 'matches' then input parameter 'pattern' is expected for the method

File: src/test/java/tests/HurtMePlenty.java [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+        String vmClass = estimateCompletePage.getVmClass();
+        Assert.assertTrue(vmClass.equalsIgnoreCase(VM_OPTION.getValue()), "Vm classes don't match!");
+    }
+
+    @Test
+    public void  InstanceTypeMatches() {
```
Comment (voljabes) [inline src/test/java/tests/HurtMePlenty.java:44]: consider to use non capital letters for method names

File: src/test/java/tests/Hardcore.java (hunk 3 above)
Comment (Truput) [reply src/test/java/tests/Hardcore.java:14]: I just named it as task is named

**Review by voljabes — APPROVED** [review]

**Pull request merged**